api/client/samples/analysis_kits/stocks_to_use_price_model/stocks_to_use_price_model.ipynb

```diff
@@ -0,0 +1,590 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "import os\n",
+    "import calendar\n",
+    "import pandas as pd\n",
+    "%matplotlib inline\n",
+    "import matplotlib.pyplot as plt\n",
+    "from matplotlib.backends.backend_pdf import PdfPages\n",
+    "from sklearn import linear_model\n",
+    "\n",
+    "from api.client.gro_client import GroClient\n",
+    "\n",
+    "API_HOST = 'api.gro-intelligence.com'\n",
+    "TOKEN    = os.environ['GROAPI_TOKEN']\n",
+    "client   = GroClient(API_HOST, TOKEN)\n",
+    "\n",
+    "# define region\n",
+    "UNITED_STATES_REGION_ID = client.search_for_entity('regions', 'United States of America')\n",
+    "\n",
+    "# define metrics\n",
+    "SETTLEMENT_PRICE_METRIC_ID = client.search_for_entity('metrics', 'futures prices settle (currency/mass)')\n",
+    "ENDING_STOCKS_METRIC_ID = client.search_for_entity('metrics', 'stocks, ending quantity (mass)') \n",
+    "CONSUMPTION_METRIC_ID = client.search_for_entity('metrics', 'domestic consumption (mass)')\n",
+    "EXPORT_METRIC_ID = client.search_for_entity('metrics', 'export volume (mass)')\n",
+    "\n",
+    "# define sources\n",
+    "CME = client.search_for_entity('sources', 'CME')\n",
+    "PSD = client.search_for_entity('sources', 'PS&D')\n",
+    "\n",
+    "crop_dictionary = {'corn':{'futures_contract_month':12},\n",
+    "                   'soybeans':{'futures_contract_month':11}}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define Data Functions\n",
+    "\n",
+    "In these functions, we standardize and organize our data. First, we create a rolling futures price series, which includes a single price for each trading day over the full history of CME/CBOT trading."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def contract_month_history(crop, contract_month):\n",
+    "    \"\"\"full history of settlement prices for a specific commodity\n",
+    "    and month of expiry. e.g. December corn\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    crop : string\n",
+    "        'corn' or 'soybeans'\n",
+    "    contract_month : integer\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    pandas.Dataframe\n",
+    "    \"\"\"\n",
+    "    market = client.search_for_entity('items', crop) \n",
+    "    \n",
+    "    client.add_single_data_series({'metric_id': SETTLEMENT_PRICE_METRIC_ID, \n",
+    "                                   'item_id': market, \n",
+    "                                   'region_id': UNITED_STATES_REGION_ID, \n",
+    "                                   'source_id': CME,\n",
+    "                                   'start_date': '2006-01-01',\n",
```

Comment:
Why 2006? Comment says it's the full history - why can't a start date be omitted?

Reply:
supply/demand data with revisions is only available starting in 2006, so I thought it was counterproductive to pull price data back to 1972